com.unity.multiplayer.mlapi/Tests/Editor/ProfilerTests.cs

```diff
@@ -0,0 +1,192 @@
+            {
+                ProfilerNotifier.OnPerformanceDataEvent += TestProfilerOnPerformanceDataEventNoTransport;
+            }
+            else
+            {
+                ProfilerNotifier.OnPerformanceDataEvent += TestProfilerOnPerformanceDataEventNormal;
+            }
+
+            EditorApplication.UnlockReloadAssemblies();
+            var testProfiler = new TestHasProfilable();
+            testProfiler.Initialize(useNullTransport);
+            return testProfiler;
+        }
+
+        [Test]
+        public void TestNormalRegisterAndNotifyFlowNull()
+        {
+            const bool useNullTransport = true;
+            TestHasProfilable testProfiler = SetupTestProfiler(useNullTransport);
+
+            TestHasProfilable.ProfilerBeginTick();
+            testProfiler.Send();
+            TestHasProfilable.NotifyProfilerListeners();
+
+            BreakDownTestProfiler(useNullTransport);
+        }
+
+        [Test]
+        public void TestNormalRegisterAndNotifyFlow()
+        {
+            const bool useNullTransport = false;
+            TestHasProfilable testProfiler = SetupTestProfiler(useNullTransport);
+
+            TestHasProfilable.ProfilerBeginTick();
+            testProfiler.Send();
+            TestHasProfilable.NotifyProfilerListeners();
+
+            BreakDownTestProfiler(useNullTransport);
+        }
+
+        [Test]
+        public void TestDroppedRegisterAndNotifyFlow()
+        {
+            const bool useNullTransport = false;
+            TestHasProfilable testProfiler = SetupTestProfiler(useNullTransport);
+
+            TestHasProfilable.ProfilerBeginTick();
+            testProfiler.Send();
+            TestHasProfilable.NotifyProfilerListeners();
+
+            // Capturing data after notifying listeners is bad
+            Assert.Catch<NoTickDataException>(() =>
+            {
+                testProfiler.Send();
+            });
+            Assert.Catch<NoTickDataException>(() =>
+            {
+                testProfiler.Send();
+            });
+            TestHasProfilable.ProfilerBeginTick();
+
+            BreakDownTestProfiler(useNullTransport);
+        }
+
+
+        [Test]
+        public void TestProperMatchRegisterAndNotifyFlow()
+        {
+            const bool useNullTransport = false;
+            TestHasProfilable testProfiler = SetupTestProfiler(useNullTransport);
+
+            TestHasProfilable.NotifyProfilerListeners();
+            TestHasProfilable.ProfilerBeginTick();
+            testProfiler.Send();
+            TestHasProfilable.NotifyProfilerListeners();
+
+            BreakDownTestProfiler(useNullTransport);
+        }
+
+        private static void TestProfilerOnPerformanceDataEventNormal(PerformanceTickData profilerData)
+        {
+            Assert.IsTrue(profilerData.HasData(TestHasProfilable.ProfilerConstants.NetworkTestData));
+            Assert.IsTrue(profilerData.HasData(TestTransport.ProfilerConstants.TransportTestData));
+        }
```

Comment:
This method is known as a custom assertion. As such, recommend renaming this to include the word Assert in the title so it's clear to the user that it can fail in that way.

For example, this type of name 

**_AssertNetworkAndTransportDataExist_**